File: testEphemeris.js
process.env.SE_EPHE_PATH = './ephe';
const swe = require('swisseph');

// Julianisches Datum für 10. Juni 1993, 10:57 Uhr UTC
const julDay = 2449143.95694;

// Ausgabeformat
function logPlanet(name, result) {
  if (!result || result.return !== 0) {
    console.error(`❌ Fehler bei ${name}:`, result?.serr || 'Unbekannter Fehler');
  } else {
    console.log(`${name} (Longitude): ${result.xx[0].toFixed(2)}°`);
  }
}

// ☀️ Sonne
logPlanet("☀️ Sonne", swe.swe_calc_ut(julDay, swe.SE_SUN, swe.SEFLG_SWIEPH));

// ☊ Mondknoten
logPlanet("☊ Mondknoten (mean)", swe.swe_calc_ut(julDay, swe.SE_MEAN_NODE, swe.SEFLG_SWIEPH));

// ⚫ Lilith
logPlanet("⚫ Lilith", swe.swe_calc_ut(julDay, swe.SE_MEAN_APOG, swe.SEFLG_SWIEPH));

File: test_ephemeris.py
import swisseph as swe

# Set path to ephemeris files
swe.set_ephe_path('./ephe')

# Julian Day for 10 June 1993, 10:57 UTC
jd = swe.julday(1993, 6, 10, 10 + 57 / 60)

# Helper to format and print planetary data
def log_planet(name, planet_id):
    try:
        values, flags = swe.calc_ut(jd, planet_id)
        lon = values[0]  # geozentrische Länge
        print(f"{name} (Longitude): {lon:.2f}°")
    except Exception as e:
        print(f"❌ Fehler bei {name}: {str(e)}")

# Run calculations
log_planet("☀️ Sonne", swe.SUN)
log_planet("☊ Mondknoten (mean)", swe.MEAN_NODE)
log_planet("⚫ Lilith", swe.MEAN_APOG)

